src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoroelasticKernel.hpp

```diff
@@ -0,0 +1,447 @@
+    real64 bodyForce[3] = { m_gravityVector[0],
+                            m_gravityVector[1],
+                            m_gravityVector[2]};
+    if( m_gravityAcceleration > 0.0 )
+    {
+      real64 mixtureDensity = ( 1.0 - porosityNew ) * m_solidDensity( k, q ) + porosityNew * m_fluidDensity( k, q );
+      mixtureDensity *= detJxW;
+      bodyForce[0] *= mixtureDensity;
+      bodyForce[1] *= mixtureDensity;
+      bodyForce[2] *= mixtureDensity;
+    }
+
+    // Assemble local jacobian and residual
+
+    // --- Momentum balance
+    for( localIndex i=0; i<6; ++i )
+    {
+      totalStress[i] *= -detJxW;
+    }
+
+    FE_TYPE::plusGradNajAijPlusNaFi( dNdX,
+                                     totalStress,
+                                     N,
+                                     bodyForce,
+                                     reinterpret_cast< real64 (&)[numNodesPerElem][3] >(stack.localResidual) );
+
+    stiffness.template upperBTDB< numNodesPerElem >( dNdX, -detJxW, stack.localJacobian );
+
+    for( integer a = 0; a < numNodesPerElem; ++a )
+    {
+      stack.localDispFlowJacobian[ a * 3 + 0][0] += dNdX[a][0] * m_biotCoefficient * detJxW;
+      stack.localDispFlowJacobian[ a * 3 + 1][0] += dNdX[a][1] * m_biotCoefficient * detJxW;
+      stack.localDispFlowJacobian[ a * 3 + 2][0] += dNdX[a][2] * m_biotCoefficient * detJxW;
+    }
+
+    if( m_gravityAcceleration > 0.0 )
+    {
+      // Assumptions: ( i) dMixtureDens_dVolStrain contribution is neglected
+      //              (ii) grains are assumed incompressible
+
+      real64 const dMixtureDens_dPressure = dPorosity_dPressure * ( -m_solidDensity( k, q ) + m_fluidDensity( k, q ) )
+                                            + porosityNew * m_dFluidDensity_dPressure( k, q );
+      for( integer a = 0; a < numNodesPerElem; ++a )
+      {
+        stack.localDispFlowJacobian[ a * 3 + 0][0] += N[a] * dMixtureDens_dPressure * m_gravityVector[0] * detJxW;
+        stack.localDispFlowJacobian[ a * 3 + 1][0] += N[a] * dMixtureDens_dPressure * m_gravityVector[1] * detJxW;
+        stack.localDispFlowJacobian[ a * 3 + 2][0] += N[a] * dMixtureDens_dPressure * m_gravityVector[2] * detJxW;
+      }
+    }
+
+    // --- Mass balance accumulation
+    for( integer a = 0; a < numNodesPerElem; ++a )
+    {
+      stack.localFlowDispJacobian[ 0][a * 3 + 0] += dPorosity_dVolStrainIncrement * m_fluidDensity( k, q ) * dNdX[a][0] * detJxW;
+      stack.localFlowDispJacobian[ 0][a * 3 + 1] += dPorosity_dVolStrainIncrement * m_fluidDensity( k, q ) * dNdX[a][1] * detJxW;
+      stack.localFlowDispJacobian[ 0][a * 3 + 2] += dPorosity_dVolStrainIncrement * m_fluidDensity( k, q ) * dNdX[a][2] * detJxW;
+    }
+
+    stack.localFlowResidual[0] += ( porosityNew * m_fluidDensity( k, q ) - porosityOld * m_fluidDensityOld( k ) ) * detJxW;
+    stack.localFlowFlowJacobian[0][0] += ( dPorosity_dPressure * m_fluidDensity( k, q ) + porosityNew * m_dFluidDensity_dPressure( k, q ) ) * detJxW;
+
+  }
```

Comment:
Can we extend these residual and jacobian calculations to THM coupling? Are there any difficulty to do so?

Reply:
I don't anticipate any difficulty. The design question I have is whether we want to have a unique single phase poroelastic solver (and its multiphase counterpart) that has all coupling we're interested in or independent solvers. Based on the last discussion we had, I understand we prefer the second alternative.